serde-generate/src/dart.rs

```diff
@@ -1218,7 +1218,21 @@ impl crate::SourceInstaller for Installer {
         let generator = CodeGenerator::new(config);
         generator.output(self.install_dir.clone(), registry)?;
         generator.output_test(&self.install_dir)?;
-        self.install_runtime(include_directory!("runtime/dart/test"), "test/src")?;
+        std::fs::create_dir_all(self.install_dir.join("test/src")).unwrap();
+
+        let tests = include_directory!("runtime/dart/test");
```

Comment:
This part still feels a bit complicated. Perhaps we can just not copy the unit tests of the runtime library (or copy them as part of the install_*_runtime methods ?)

Reply:
I'm happy to defer to your preference on what we should do with unit tests. In a way it feels unnecessary to me to install tests into the end user's project. But there was logic like this present previously so I kept it and just modified to only install the enabled encodings. I'm working on this today so can pull it out right now if you like.

Reply:
Looking at some of the other implementations I think my pick would be to get rid of Dart's `output_test` and drop this test copying logic from `install_module`, then copy the individual files in the `dart_runtime.rs` tests. Thoughts?

Reply:
Here's what that would look like https://github.com/jerel/serde-reflection/commit/d46f6bdde9b0b269afc6a1567cec1b0f854975a7 If you like that approach I can add it to this PR.

Reply:
Yes please let's not generate or copy test during codegen. We do run unit tests as part of the `xx_runtime.rs` tests files under a separate test often called `xx_autotest` or equivalent

Reply:
Ok, the above commit is added to the PR but as `test_dart_runtime_autotest` instead of running with the encoding runtime tests. I'm finished with this PR unless there's something else that you see.